python/ray/util/state/state_cli.py

```diff
@@ -0,0 +1,1308 @@
+
+        ```
+        ray logs [cluster] raylet.out --tail 500 —-node-id A
+        ```
+
+        Download the gcs_server.txt file to the local machine.
+
+        ```
+        ray logs [cluster] gcs_server.out --tail -1 > gcs_server.txt
+        ```
+
+        Follow the log files from the last 100 lines.
+
+        ```
+        ray logs [cluster] raylet.out --tail 100 -f
+        ```
+
+    Raises:
+        :class:`RayStateApiException <ray.util.state.exception.RayStateApiException>` if the CLI
+            is failed to query the data.
+    """  # noqa: E501
+
+    if node_id is None and node_ip is None:
+        node_ip = _get_head_node_ip(address)
+
+    logs = list_logs(
+        address=address,
+        node_id=node_id,
+        node_ip=node_ip,
+        glob_filter=glob_filter,
+        timeout=timeout,
+    )
+
+    log_files_found = []
+    for _, log_files in logs.items():
+        for log_file in log_files:
+            log_files_found.append(log_file)
+
+    if len(log_files_found) != 1:
+        # Print the list of log files found if no unique log found
+        if node_id:
+            print(f"Node ID: {node_id}")
+        elif node_ip:
+            print(f"Node IP: {node_ip}")
+        print(output_with_format(logs, schema=None, format=AvailableFormat.YAML))
+        return
+
+    # If there's only 1 file, that means there's a unique match.
+    filename = log_files_found[0]
+
+    _print_log(
+        address=address,
+        node_id=node_id,
+        node_ip=node_ip,
+        filename=filename,
+        tail=tail,
+        follow=follow,
+        interval=interval,
+        timeout=timeout,
+        encoding=encoding,
+        encoding_errors=encoding_errors,
+    )
+
+
+@logs_state_cli_group.command(name="actor")
+@click.option(
+    "--id",
+    "-a",
+    required=False,
+    type=str,
+    default=None,
+    help="Retrieves the logs corresponding to this ActorID.",
+)
+@click.option(
+    "--pid",
+    "-pid",
+    required=False,
+    type=str,
+    default=None,
+    help="Retrieves the logs from the actor with this pid.",
+)
+@address_option
+@log_node_id_option
+@log_node_ip_option
+@log_follow_option
+@log_tail_option
+@log_interval_option
+@log_timeout_option
+@log_suffix_option
+@click.pass_context
+@PublicAPI(stability="stable")
+def log_actor(
+    ctx,
+    id: Optional[str],
+    pid: Optional[str],
+    address: Optional[str],
+    node_id: Optional[str],
+    node_ip: Optional[str],
+    follow: bool,
+    tail: int,
+    interval: float,
+    timeout: int,
+    err: bool,
+):
+    """Get/List logs associated with an actor.
+
+    Example:
+
+        Follow the log file with an actor id ABC.
+
+        ```
+        ray logs actor --id ABC --follow
+        ```
+
+        Get the actor log from pid 123, ip ABC.
+        Note that this goes well with the driver log of Ray which prints
+        (ip=ABC, pid=123, class_name) logs.
+
+        ```
+        ray logs actor --pid=123  —ip=ABC
```

Comment:
Is this a typo (-ip -> --ip)?

Also I'm slightly confused that you are using `ip=ABC` here and above and then below you are using `--id ABC` -- for the id it might be better to use something different than ABC so users don't confuse it with the ip from above (e.g. you could use an example actor ID).

Reply:
> Is this a typo (-ip -> --ip)?
I think we are using --node-ip for long flag, and `-ip` for short flag. 


Updated with better names!